tests/freshdesk.test.ts

```diff
@@ -240,4 +240,140 @@ describe('generateFreshdeskTicketFieldDefinitions', () => {
       },
     ])
   })
+
+  it('should return the correct value when choices is used over options', () => {
+    const fields: FreshdeskTypes.FreshdeskField[] = [
+      //@ts-expect-error Not filling in all properties
+      {
+        id: 3,
+        name: 'Field3',
+        type: 'custom_dropdown',
+        label_for_customers: 'Field 3',
+        required_for_customers: true,
+        choices: ['value 1', 'value 2'],
+      },
+    ]
+    expect(generateFreshdeskTicketFieldDefinitions(fields)).toEqual([
+      {
+        checkIsFormElementSupported: expect.any(Function),
+        choices: [
+          {
+            label: 'value 1',
+            value: 'value 1',
+          },
+          {
+            label: 'value 2',
+            value: 'value 2',
+          },
+        ],
+        displayName: 'Field 3',
+        freshdeskFieldName: 'Field3',
+        id: '3',
+        isRequired: true,
+        type: 'CHOICE_SINGLE',
+      },
+    ])
+  })
+
+  it('should map DEPENDENT_FIELD field correctly when choices is used', () => {
+    const fields: FreshdeskTypes.FreshdeskField[] = [
+      //@ts-expect-error Not filling in all properties
```

Comment:
Can we put our freshdesk fields here instead of just some fields.

Reply:
To clarify: are you saying to put all of our freshdesk fields into this one test?

Reply:
Yes

Reply:
It seems a bit overkill to have all 16 different fields here for the test. I think it would be better to have the test include a mixture of fields options + choices and one without choices to ensure they all come out correct. Is there a reason to have all the freshdesk fields?

Reply:
Pass them all in, check snapshot value, regressions are now impossible.

Could pass in individual ones if you want but a lot more work for the same result and more maintenance.

Reply:
In the same test, run the function again with the result as the input and assert the input is the same as the output. That will cover all scenarios.

Reply:
Seems like this scenario would be better as a new test as opposed to including in this test. Then, we have a test that ensures the functionality of the mapping and we have a test that ensures the fields are transforming correctly.